lightning/src/ln/channel.rs

```diff
@@ -3525,6 +3573,13 @@ impl<Signer: Sign> Channel<Signer> {
 						                             ((index_in_block as u64) << (2*8)) |
 						                             ((txo_idx as u64)        << (0*8)));
 					}
+					// If we allow 1-conf funding, we may need to check for funding_locked here and
+					// send it immediately instead of waiting for an update_best_block call (which
+					// may have already happened for this block).
+					// XXX: Test this case!
+					if let Some(funding_locked) = self.check_get_funding_locked(height) {
+						return Ok(Some(funding_locked));
```

Comment:
We could, but we don't document which message events a function may generate in any other place? What makes this site special - in general the PeerHandler should just be polling for new message events regularly.

Reply:
I think it would serve more as a friendly note for library developers than library users. It was surprising at first sight with the old API model in mind, now I get it.